pytorch_lightning/trainer/trainer.py

```diff
@@ -1431,58 +1437,109 @@ def _call_teardown_hook(self) -> None:
-                accelerator_hook = getattr(self.accelerator, hook_name)
-                accelerator_output = accelerator_hook(*args, **kwargs)
-                # Rely on the accelerator output if lightningModule hook returns nothing
-                # Required for cases such as DataParallel where we reduce the output for the user
-                # todo: move this data parallel logic into the data parallel plugin
-                output = accelerator_output if output is None else output
-
-            # call the ttp hook
-            if hook_name not in ("setup", "teardown", "on_train_start") and hasattr(
-                self.training_type_plugin, hook_name
-            ):
-                ttp_hook = getattr(self.training_type_plugin, hook_name)
-                ttp_output = ttp_hook(*args, **kwargs)
-                output = ttp_output if output is None else output
+        # TODO: remove first if statement in v1.7
+        if hook_name in ("on_train_batch_start", "on_train_batch_end"):
+            fn = getattr(self, hook_name)
+            if callable(fn):
+                with self.profiler.profile(hook_name):
+                    output = fn(*args, **kwargs)
+        elif hook_name in ("on_init_start", "on_init_end"):
+            # these `Callback` hooks are the only ones that do not take a lightning module.
+            # we also don't profile bc profiler hasn't been set yet
+            for callback in self.callbacks:
+                fn = getattr(callback, hook_name)
+                if callable(fn):
+                    output = fn(self, *args, **kwargs)
+        else:
+            for callback in self.callbacks:
+                fn = getattr(callback, hook_name)
+                if callable(fn):
+                    with self.profiler.profile(hook_name):
+                        output = fn(self, self.lightning_module, *args, **kwargs)
 
         if pl_module:
             # restore current_fx when nested context
             pl_module._current_fx_name = prev_fx_name
 
         return output
 
+    # TODO: eventually no longer need this
+    def _call_ttp_hook(
+        self,
+        hook_name: str,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        output = None
+        fn = getattr(self.training_type_plugin, hook_name)
+        if not callable(fn):
+            raise MisconfigurationException("Hook is not callable")
+
+        with self.profiler.profile(hook_name):
+            output = fn(*args, **kwargs)
+
+        return output
+
+    # TODO: eventually no longer need this
+    def _call_accelerator_hook(
+        self,
+        hook_name: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Optional[Any]:
+        output = None
+        fn = getattr(self.accelerator, hook_name)
+        if not callable(fn):
+            raise MisconfigurationException("Hook is not callable")
+
+        with self.profiler.profile(hook_name):
+            output = fn(*args, **kwargs)
+
+        return output
```

Comment:
@four4fish @awaelchli  fyi since you are actively refactoring the accelerator/training type plugin structure. this should go away after that refactor

Reply:
Will be call_strategy_hook right?

Reply:
yes, we can rename the `_call_ttp_hook` to `_call_strategy_hook` once the rename is formalized, and remove this `_call_accelerator_hook` entirely

Reply:
Issue created https://github.com/PyTorchLightning/pytorch-lightning/issues/10905

Reply:
yup, I am planning on doing this as a follow up to this PR